Yank/multistate/sams.py

```diff
@@ -433,35 +434,37 @@ def _local_jump(self, jump_and_mix_data):
             new_state_index = np.random.choice(neighborhood, p=P_k[neighborhood])
             self._replica_thermodynamic_states[replica_index] = new_state_index
             # Accumulate statistics
-            jump_and_mix_data.replica_log_P_k[replica_index,:] = log_P_k[:]
+            replicas_log_P_k[replica_index,:] = log_P_k[:]
             self._n_proposed_matrix[current_state_index, neighborhood] += 1
             self._n_accepted_matrix[current_state_index, new_state_index] += 1
 
-    def _global_jump(self, jump_and_mix_data):
+    def _global_jump(self, replicas_log_P_k):
         """
         Global jump scheme.
         This method is described after Eq. 3 in [2]
         """
         n_replica, n_states = self.n_replicas, self.n_states
-        for (replica_index, current_state_index) in enumerate(self._replica_thermodynamic_states):
-            u_k = np.zeros([n_states], np.float64)
-            log_P_k = np.zeros([n_states], np.float64)
-            # Compute unnormalized log probabilities for all thermodynamic states
+        for replica_index, current_state_index in enumerate(self._replica_thermodynamic_states):
             neighborhood = self._neighborhood(current_state_index)
+
+            # Compute unnormalized log probabilities for all thermodynamic states.
+            log_P_k = np.zeros([n_states], np.float64)
             for state_index in neighborhood:
-                u_k[state_index] = self._energy_thermodynamic_states[replica_index, state_index]
-                log_P_k[state_index] = self.log_weights[state_index] - u_k[state_index]
+                u_k = self._energy_thermodynamic_states[replica_index, state_index]
```

Comment:
To match the equations, it would be clearer to write:
```python
u_k = self._energy_thermodynamic_states[replica_index, :]
log_P_k[state_index] = - u_k[state_index] + self.log_weights[state_index] 
```

Reply:
Good idea! Will change.